puma/integrated_eff.py

```diff
@@ -0,0 +1,269 @@
+        """IntegratedEfficiency plot properties.
+
+        Parameters
+        ----------
+        grid : bool, optional
+            Set the grid for the plots.
+        **kwargs : kwargs
+            Keyword arguments from `puma.PlotObject`
+        """
+        super().__init__(grid=grid, **kwargs)
+        self.int_effs = {}
+        self.tagger_ls = {}
+        self.label_colours = {}
+        self.leg_tagger_labels = {}
+        self.initialise_figure()
+        self.disc_min, self.disc_max = (1e3, -1e3)
+        self.default_linestyles = get_good_linestyles()
+        self.legend_flavs = None
+        self.leg_tagger_loc = "lower left"
+
+        self.ymin = 0
+        self.ymax = 1.2
+
+    def add(self, int_eff: object, key: str = None):
+        """Adding puma.Roc object to figure.
+
+        Parameters
+        ----------
+        int_effs : puma.IntegratedEfficiency
+            IntegratedEfficiency curve
+        Raises
+        ------
+        KeyError
+            If unique identifier key is used twice
+        """
+        if key is None:
+            key = len(self.int_effs) + 1
+        if key in self.int_effs:
+            raise KeyError(
+                f"Duplicated key {key} already used for roc unique identifier."
+            )
+
+        self.int_effs[key] = int_eff
+        # set linestyle
+        if int_eff.tagger not in self.tagger_ls:
+            self.tagger_ls[int_eff.tagger] = (
+                self.default_linestyles[len(self.tagger_ls)]
+                if int_eff.linestyle is None
+                else int_eff.linestyle
+            )
+        elif (
+            int_eff.linestyle != self.tagger_ls[int_eff.tagger]
+            and int_eff.linestyle is not None
+        ):
+            logger.warning(
+                (
+                    "You specified a different linestyle for the same tagger"
+                    " %s. This will lead to a mismatch in the line colours"
+                    " and the legend."
+                ),
+                int_eff.tagger,
+            )
+        if int_eff.linestyle is None:
+            int_eff.linestyle = self.tagger_ls[int_eff.tagger]
+
+        # set colours
+        if int_eff.label not in self.label_colours:
+            self.label_colours[int_eff.label] = (
+                get_good_colours()[len(self.label_colours)]
+                if int_eff.colour is None
+                else int_eff.colour
+            )
+        elif (
+            int_eff.colour != self.label_colours[int_eff.label]
+            and int_eff.colour is not None
+        ):
+            logger.warning(
+                (
+                    "You specified a different colour for the same label"
+                    " %s. This will lead to a mismatch in the line colours"
+                    " and the legend."
+                ),
+                int_eff.label,
+            )
+        if int_eff.colour is None:
+            int_eff.colour = self.label_colours[int_eff.label]
```

Comment:
I think the control plots have the wrong colour, no? It should have the pre-defined colours for the classes

Reply:
Maybe it's not a good solution to pre-define classes (like b-, c-, light-jets) because someone may want to use it for other jet types (in Xbb tagging, for example).
Or do you think that it is better to fix the possible types?

Reply:
If there is a simple way of checking if the flavour exists, we should use the defined colour. We have definitions for the Xbb labelling and other extended labelling, so in most cases it will make sense to reuse this. If the flavour is not defined, we can just let the colour be set automatically

Reply:
I added the `flavour` argument, so if it defined the colour will be selected based on this